music21/midi/translate.py

```diff
@@ -267,7 +269,60 @@ def music21ObjectToMidiFile(
 # ------------------------------------------------------------------------------
 # Notes
 
-def midiEventsToNote(eventList, ticksPerQuarter=None, inputM21=None) -> note.Note:
+def _constructOrUpdateNotRest(
+    eOn: 'music21.midi.MidiEvent',
+    tOn: int,
+    tOff: int,
+    ticksPerQuarter: int,
+    *,
+    inputM21: Optional[note.NotRest] = None,
+    preferredClass: Optional[Callable] = note.Note,
+) -> note.NotRest:
+    '''
+    Construct (or edit the duration of) a NotRest subclass, usually
+    a note.Note (or a chord.Chord if provided to `preferredClass`).
+
+    If the MidiEvent is on channel 10, then an Unpitched or PercussionChord
+    is constructed instead. Raises TypeError if an incompatible object is provided
+    for `inputM21`, e.g. a `chord.Chord` when a `percussion.PercussionChord` is needed.
+    '''
+    if not isinstance(preferredClass(), note.NotRest):
+        raise TypeError(f'Expected subclass of note.NotRest; got {preferredClass}')
+
+    if eOn.channel == 10:
+        if preferredClass in (chord.Chord, percussion.PercussionChord):
+            preferredClass = percussion.PercussionChord
+        else:
+            preferredClass = note.Unpitched
+
+    if inputM21 is not None:
+        if not isinstance(inputM21, preferredClass):
```

Comment:
if we have other PRs open to speeding up MIDI, is this isinstance check needed for the rare case that someone gives an incorrect class?  It's now a private routine.